sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -325,6 +325,24 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       Row(6))
   }
 
+  test("sorting with null ordering") {
+    val data = Seq[java.lang.Integer](2, 1, null).toDF("key")
```

Comment:
Thanks for the cleanup and refining the test case! For the test case with 2 column data, I was thinking to create a bit more complex ordering to utilize the prefix sort and regular `GenerateOrdering` in one query. But it is not necessary since the major PR for SQL interface has include such scenario.